src/flip.jl

```diff
@@ -185,7 +185,7 @@ function flip_chain(
                     continue
                 end
                 # Repeat the last scores upon rejection (when self-loops are enabled).
-                push!(chain_scores.step_values, copy(chain_scores.step_values[end]))
+                score_vals = copy(chain_scores.step_values[end])
```

Comment:
if someone's doing analysis on the chain later, each element of the chain scores array should correspond to the partition, including self loops, so that implies that we need to copy the scores, right?

Reply:
Ah ok, that makes sense. If `copy` is shallow, then this looks fine -- my own concern was a potential performance penalty.